components/ingest-service/projectupdater/manager.go

```diff
@@ -25,88 +26,126 @@ const (
-	return Manager{
-		state:               notRunningState,
+	eventServiceClient automate_event.EventServiceClient, configManager *config.Manager) *Manager {
+
+	updateQueue := make(chan func(stage) stage, 100)
+	manager := &Manager{
+		stage: stage{
+			state: notRunningState,
+		},
 		client:              client,
 		authzProjectsClient: authzProjectsClient,
 		eventServiceClient:  eventServiceClient,
+		configManager:       configManager,
+		updateQueue:         updateQueue,
 	}
+
+	// Single goroutine that updates the stage data
+	go manager.stageUpdater(updateQueue)
+
+	manager.resumePreviousState()
+	return manager
 }
 
+// Cancel - stop any running job
+// Action
 func (manager *Manager) Cancel(projectUpdateID string) {
-	switch manager.state {
-	case notRunningState:
-		// do nothing job is not running
-	case runningState:
-		if manager.projectUpdateID == projectUpdateID {
+	manager.updateStage(func(stage stage) stage {
+		switch manager.stage.state {
+		case notRunningState:
+			// do nothing job is not running
+		case runningState:
+			if manager.stage.projectUpdateID != projectUpdateID {
+				// do nothing because the requested project update job is not running
+				return stage
+			}
 			logrus.Debugf("Cancelling project tag update for ID %q elasticsearch task ID %q",
-				manager.projectUpdateID, manager.esJobID)
-			manager.client.JobCancel(context.Background(), manager.esJobID)
-		} else {
-			// do nothing because the requested project update job is not running
+				manager.stage.projectUpdateID, manager.stage.esJobID)
+			manager.client.JobCancel(context.Background(), manager.stage.esJobID)
+		default:
+			// error state not found
+			manager.sendFailedEvent(fmt.Sprintf(
+				"Internal error state %q eventID %q", manager.stage.state, projectUpdateID),
+				projectUpdateID)
 		}
-	default:
-		// error state not found
-		manager.sendFailedEvent(fmt.Sprintf(
-			"Internal error state %q eventID %q", manager.state, manager.projectUpdateID),
-			projectUpdateID)
-	}
+
+		return stage
+	})
 }
 
 // Start - start a project update
+// Action
 func (manager *Manager) Start(projectUpdateID string) {
-	switch manager.state {
-	case notRunningState:
-		// Start elasticsearch update job async and return the job ID
-		esJobID, err := manager.startProjectTagUpdater()
-		if err != nil {
-			logrus.Errorf("Failed to start Elasticsearch Project rule update job projectUpdateID: %q", projectUpdateID)
-			manager.sendFailedEvent(fmt.Sprintf(
-				"Failed to start Elasticsearch Project rule update job projectUpdateID: %q", projectUpdateID),
-				projectUpdateID)
-			return
-		}
-		manager.esJobID = esJobID // Store the job ID and event ID
-		manager.projectUpdateID = projectUpdateID
-		manager.state = runningState
-		go manager.waitingForJobToComplete()
-	case runningState:
-		if manager.projectUpdateID == projectUpdateID {
-			//	Do nothing. The job has ready started
-		} else {
+	manager.updateStage(func(stage stage) stage {
```

Comment:
This adds this function to the `updateQueue` to be processed in order by a single thread.